### PR TITLE
Added example dockerfile for using MySQL as datastore and updated doc…

### PR DESCRIPTION
**Pull request by code040**
…umentation

There is a main Dockerfile, creating an image to run a 'bare' OpenTAXII server (using SQLite for database).  The example Dockerfile uses a MySQL datastore, and sets up a fully running, and configured instance.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Set the base image to Python
+FROM python:2.7.9
+MAINTAINER code040@intelworks.com
```
Comment (traut) [inline Dockerfile]: maybe use our default email here for consistency? https://github.com/Intelworks/OpenTAXII/blob/master/setup.py#L19

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# Install dependencies
+COPY ./requirements.txt /requirements.txt
+RUN pip install -r requirements.txt \
+    && pip install supervisor \
+    && pip install gunicorn
+
+# Install OpenTaxii
+RUN mkdir /opentaxii
+COPY ./ /opentaxii/
+RUN cd  /opentaxii && python setup.py install && rm -rf /opentaxii
```
Comment (traut) [inline Dockerfile]: does it make sense to create virtualenv here? I don't know much about how Docker works

File: docs/docker.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+To get a default (development) instance using docker
+
+
+.. code-block:: shell
+
+    docker run -d -p 9000:9000 intelworks/opentaxii
+
+.. note::
+
+    OpenTAXII is now accessible through port 9000, and is not configured.
```
Comment (traut) [inline docs/docker.rst]: should we provide default configuration? Maybe one service per type, one `default` collection. What do you think?

File: Dockerfile (hunk 1 above)
Comment (code040) [inline Dockerfile]: yep. will do

File: Dockerfile (hunk 2 above)
Comment (code040) [inline Dockerfile]: not really needed, because an image has a single purpose. To use virtualenv makes it too convoluted.

**Comment by traut** [comment]
I think it will be tidier with a new directory called `docker` in / and all docker-related stuff there

File: docs/docker.rst (hunk 3 above)
Comment (code040) [inline docs/docker.rst]: In the examples it has that. But here it would make sense to add as well. But then leave out a user?

**Comment by code040** [comment]
The Dockerfile needs to be relative to the resources it needs. So it will not be able move into separate directory, but some of the resources could move.

**Comment by traut** [comment]
Right, `Dockerfile` is like a `Makefile`, it can stay in `/` for sure

File: docs/docker.rst (hunk 3 above)
Comment (traut) [inline docs/docker.rst]: Yeah, we can leave out the user (then services need to be configured with authentication set to false) or configure a default one (test/test). I think a default user is nicer - shows auth features as well

File: Dockerfile (hunk 2 above)
Comment (traut) [inline Dockerfile]: got it

File: docs/docker.rst (hunk 3 above)
Comment (code040) [inline docs/docker.rst]: Hmm .. Ok. will make the base a bit more ‘complete’, and move it out of the ‘example’ one.

**Pull request merged**